### ISSUE TITLE
Add Model Epoch vs Loss Graphs to Visualize Model Training Performance

### ISSUE THREAD

**Issue by PriyanshuRao-code** [PriyanshuRao-code/AI-Lab-Project#8]
Most of the models (especially iterative ones like Perceptron, Logistic Regression, Neural Networks etc.) internally try to minimize loss during training across several iterations/epochs.

Visualizing how the loss decreases (or behaves) over the epochs can give useful insights like:
How fast model is learning?
Is it overfitting/underfitting?
Is loss stuck somewhere?
Are there fluctuations in loss?

TASK:
Try to generate and add "Epoch vs Loss" graphs for models which internally support or provide this functionality.
